src/GoogleMap.js

```diff
@@ -61,23 +61,30 @@ const updaterMap = {
     map.setZoom(zoom)
   }
 }
-
+const defatulMapId = "defaultMapId";
 export class GoogleMap extends PureComponent {
   static propTypes = GoogleMapPropTypes
   static defaultProps = {
-    onLoad: () => {}
+    onLoad: () => { }
   }
-
-  state = {
-    map: null
+  get mapId() {
+    return this.props.id || defatulMapId;
+  }
+  constructor(props) {
```

Comment:
We're using class properties here. So no need for a constructor. state can stay outside as it was and getInstance can be written as an arrow function (instead of binding)